### DISCUSSION TITLE
How to pass VAULT_CLIENT_CERT

### DISCUSSION THREAD

**Discussion by nbari** [ansible-collections/community.hashi_vault#402] [Q&A]
Hi, I need to pass a certificate in order to connect to vault, from the CLIi I use `VAULT_CLIENT_CERT`, but to pass the cert when using `community.hashi_vault.vault_pki_generate_certificate`

**Comment by briantist** [comment]
Hi @nbari , I believe you'd set `cert_auth_public_key` (https://docs.ansible.com/ansible/devel/collections/community/hashi_vault/vault_pki_generate_certificate_module.html#parameter-cert_auth_public_key) if you're using `auth_method=cert`.

I don't use client cert auth in Vault so it's actually something I'm a bit unsure about.

**Comment by nbari** [comment]
Hi @briantist, I am using `auth_method=token` but can't find how to send the certificate, mainly because  I have vault "behind" a proxy that requests a certificate (without the proxy works), when using  the `vault` CLI works, but I need to specify:

```
export VAULT_ADDR=https://vault.domain
export VAULT_TOKEN= 
export VAULT_CLIENT_CERT=/path/to/crt.pem
export VAULT_CLIENT_KEY=/path/to/crt.key
```
Then I can interact with it using `vault ....`.
